cmake/LibeventConfig.cmake.in

```diff
@@ -1,18 +1,178 @@
+                "The valid components are event_core, event_extra, event_openssl${pthreadlib} and event.")
+    else()
+        message_if_needed(WARNING "Your libevent installation does not contain a ${_comp} library target!")
+    endif()
+endmacro()
+
+macro(message_if_needed _flag _msg)
+    if (NOT ${CMAKE_FIND_PACKAGE_NAME}_FIND_QUIETLY)
+        message(${_flag} "${_msg}")
+    endif()
+endmacro()
+
+if(CONFIG_FOR_INSTALL_TREE)
+## Config for install tree ----------------------------------------
+# Find includes
+unset(_event_h CACHE)
+find_path(_event_h
+          NAMES event2/event.h
+          PATHS "${_INSTALL_PREFIX}/include"
+          NO_DEFAULT_PATH)
+if(_event_h)
+    set(LIBEVENT_INCLUDE_DIRS "${_event_h}")
+    message_if_needed(STATUS "Found libevent include directory: ${_event_h}")
+else()
+    message_if_needed(WARNING "Your libevent installation does not contain header files!")
+endif()
+
+# Find libraries
+macro(find_event_lib _comp)
+    unset(_event_lib CACHE)
+    find_library(_event_lib
+                NAMES ${_comp}
+                PATHS "${_INSTALL_PREFIX}/lib"
+                NO_DEFAULT_PATH)
+    if(_event_lib)
+        list(APPEND LIBEVENT_LIBRARIES ${_comp}_${_LIB_TYPE})
+        set_case_insensitive_found("${_comp}")
+        message_if_needed(STATUS "Found libevent component: ${_event_lib}")
+    else()
+        no_component_msg(${_comp})
+    endif()
+endmacro()
+
+foreach(comp ${_EVENT_COMPONENTS})
+    find_event_lib(${comp})
+endforeach()
+else()
+## Config for build tree ----------------------------------------
+set(LIBEVENT_INCLUDE_DIRS "@EVENT__INCLUDE_DIRS@")
+foreach(_comp ${_EVENT_COMPONENTS})
+    list(FIND _AVAILABLE_LIBS ${_comp} _INDEX)
+    if(_INDEX GREATER -1)
+        list(APPEND LIBEVENT_LIBRARIES ${_comp}_${_LIB_TYPE})
+        set_case_insensitive_found(${_comp})
+    else()
+        no_component_msg(${_comp})
+    endif()
+endforeach()
+
+endif()
+
+set(LIBEVENT_INCLUDE_DIR ${LIBEVENT_INCLUDE_DIRS})
+if(LIBEVENT_LIBRARIES)
+    set(LIBEVENT_LIBRARY ${LIBEVENT_LIBRARIES})
+    if(CONFIG_FOR_INSTALL_TREE)
+        message_if_needed(STATUS "Found libevent ${LIBEVENT_VERSION} in ${_INSTALL_PREFIX}")
+    else()
+        message_if_needed(STATUS "Found libevent ${LIBEVENT_VERSION} in ${LIBEVENT_CMAKE_DIR}")
+    endif()
+
+    # Avoid including targets more than one times
+    if(NOT TARGET event_core_${_LIB_TYPE})
+        # Include the project Targets file, this contains definitions for IMPORTED targets.
+        include(${LIBEVENT_CMAKE_DIR}/LibeventTargets.cmake)
+    endif()
+else()
+    if(${CMAKE_FIND_PACKAGE_NAME}_FIND_REQUIRED)
+        message(FATAL_ERROR "Can not find any libraries for libevent.")
+    else()
+        message_if_needed(WARNING "Can not find any libraries for libevent.")
+    endif()
+endif()
+
+unset(_LIB_TYPE)
+unset(_AVAILABLE_LIBS)
+unset(_EVENT_COMPONENTS)
+unset(_POSSIBLE_PKG_NAMES)
+unset(_INSTALL_PREFIX)
```

Comment:
It will be nice to have a test for this on our CI (with build-tree and install-tree and system-wide install)

Reply:
This function is difference from others. For testing, should I install the products and create a standalone project?

Reply:
>should I install the products

All available combinations should be covered:
- install into system-wide path
- install into non-system-wide path
- build tree

> and create a standalone project?

I would say that this should be in libevent tree, since otherwise it will require extra work to make it run for each commit (i.e. `sample/libevent-export-test` or something similar, also this will give an example for users)